### PR TITLE
[3/n][gql-performance] add min_tx_sequence_number and max_tx_sequence_number to checkpoints table

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

Per title - add `min_tx_sequence_number` and `max_tx_sequence_number` to `checkpoints` table to support queries mapping `cp` to a `tx` range. This is so that when given a `checkpoint_sequence_number` range, we can map it to the corresponding `tx_sequence_number` range to facilitate transactions queries.

The two new columns are nullable, because it is possible for a checkpoint to not contain any transactions. In practice, we'll typically have at least a commit prologue txn. The nullable columns do not impact the queries we intend to run on the table. If through a CTE, a null value will eventually resolve to FALSE when we attempt to use the min or max tx sequence number in a query against the transactions and tx lookup tables. If we make multiple roundtrips, we can stop early if we see that the range is empty for a checkpoint. 

## Test plan 

Existing tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Comment by amnn** [comment]
> It was very tempting to just stick this into the existing persist_checkpoints implementation, but figured it'd be better to treat it as a separate table, as we do with tables today.

I actually think you should give in to your earlier temptation -- what's the rationale for keeping it as a separate table?

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (gegaowp) [inline crates/sui-indexer/src/handlers/checkpoint_handler.rs:385]: nit: tx_seq_num_iter.fold to get (min_cp, max_cp) ?

**Review by gegaowp — COMMENTED** [review]
neat! not approving simply b/c of breaking change

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (wlmyng) [reply crates/sui-indexer/src/handlers/checkpoint_handler.rs:386]: since we're already looping through transactions once, i figured we could just plug into the existing logic instead of doing another pass, but not beholden to it

File: crates/sui-indexer/src/handlers/committer.rs
Comment (amnn) [inline crates/sui-indexer/src/handlers/committer.rs:1]: Similarly, should no longer need these changes now that `cp_tx`'s fields have been merged into `checkpoints` -- `IndexedCheckpoint` contains the same fields, so we shouldn't need to include `cp_tx` in `CheckpointDataToCommit`.

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (amnn) [inline crates/sui-indexer/src/handlers/checkpoint_handler.rs:315]: I think this is the only place where a data structure like `CpToTx` might be helpful, but if that's the case, simply passing the two sequence numbers into `from_sui_checkpoint` probably suffices.

**Review by amnn — COMMENTED** [review]
Thanks for folding it in! I think we can simplify the changes a little bit more (get rid of `CpToTx`), but otherwise this looks good. @gegaowp is right that we should target this stack to land to `idx-breaking-changes-park` rather than `main` -- I've left a comment about this on the base of the stack, so accepting to unblock on this PR.

File: crates/sui-graphql-rpc/src/server/compatibility_check.rs
Comment (amnn) [inline crates/sui-graphql-rpc/src/server/compatibility_check.rs:1]: This seems to be the old compatibility check (i.e. before my recent changes). I think the correct thing here is to not include this file in the breaking change parking branch, but rebase that branch before we start backfill.

File: crates/sui-graphql-rpc/src/types/transaction_block.rs
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/transaction_block.rs:1]: There's maybe something funky going on with the rebase, because these changes shouldn't be part of this PR -- maybe this PR needs to be rebased after the previous PR got landed?

**Review by amnn — COMMENTED** [review]
My comments may be an artifact of this PR needing to be rebased after its predecessor in the stack landed, if so, feel free to ignore!

**Comment by wlmyng** [comment]
@amnn apologies for the churn! Rebased onto `idx-breaking-change-park`. From testing, the nullable columns should not impact the performance of our intended queries. It's akin to selecting a sequence_number that doesn't exist in a checkpoints table with non-nullable tx range columns.

File: crates/sui-indexer/src/schema/pg.rs
Comment (emmazzz) [inline crates/sui-indexer/src/schema/pg.rs:326]: I'm a bit worried about the removal of this one. Does indexer reader use this for json rpc?

Comment (wlmyng) [reply crates/sui-indexer/src/schema/pg.rs:326]: Just in `indexer_reader` as a raw query, which I can address in this PR

**Review by emmazzz — APPROVED** [review]

**Pull request merged**
